src/crimson/os/seastore/cached_extent.h

```diff
@@ -531,6 +571,37 @@ class CachedExtent
     return ptr.has_value();
   }
 
+  /// Return true if range (offset -> offset + length) is loaded or is about to
+  /// be loaded (call wait_io() in this case)
+  bool is_range_loaded(extent_len_t offset, extent_len_t length) {
+    return is_fully_loaded() || (buffer_space.has_value()
+      && buffer_space->check_buffer(offset, length));
+  }
+
+  /// Check bufferspace and return uncovered interval(to load)
+  region_list_t get_unloaded_ranges(
+    extent_len_t offset, extent_len_t length) {
+    if (!buffer_space.has_value()) {
+      region_list_t ret;
+      ret.emplace_back(offset, length);
+      return ret;
+    }
+    return buffer_space->get_unloaded_intervals(offset, length);
```

Comment:
nit, less confusing if the terms can be consistent: `interval` vs `range` vs `region`